File: src/main/java/bridje/analyser/LocalVar.java
package bridje.analyser;

import bridje.runtime.Symbol;

public final class LocalVar {

    public final Symbol sym;

    public LocalVar(Symbol sym) {
        this.sym = sym;
    }

    @Override
    public String toString() {
        return String.format("(LocalVar %s@%d)", sym, hashCode());
    }
}

File: language/src/main/kotlin/brj/analyser/LocalVar.kt
package brj.analyser

import brj.runtime.Symbol

class LocalVar(val name: Symbol, val slot: Int)

sealed interface CaptureSource
data class FrameSlotCapture(val slot: Int) : CaptureSource
data class TransitiveCapture(val captureIndex: Int) : CaptureSource

data class CapturedVar(val name: Symbol, val outerLocalVar: LocalVar, val captureIndex: Int, val source: CaptureSource)
